pkg/plugins/policies/meshhttproute/plugin/v1alpha1/gateway.go

```diff
@@ -37,7 +37,12 @@ func generateGatewayListeners(
 	if info.Listener.CrossMesh {
 		protocol = mesh_proto.MeshGateway_Listener_HTTPS
 	}
-	res, filterChainBuilders, err := FilterChainGenerators[protocol].Generate(ctx, info, gatewayHosts)
+	filterGen, found := FilterChainGenerators[protocol]
+	if !found {
```

Comment:
Is this the gRPC thing or why is this necessary?

Reply:
Yes, I had a nil pointer in case of unknown protocol